lib/rubocop/cop/style/redundant_self.rb

```diff
@@ -97,8 +97,7 @@ def on_send(node)
           return unless regular_method_call?(node)
 
           return if @allowed_send_nodes.include?(node) ||
-                    @local_variables.include?(method_name)
-
+                    Array(@local_variables_scopes[node]).include?(method_name)
```

Comment:
If the previous suggestion of using `Hash#new` is adopted, this array coercion is not needed. 😀